<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name="sequence_id"                         default="022" />
    <arg name="calib_file"                          default="$(find viso_pose)/param/kitti_raw.yaml"/>
    <arg name="cam0_topic"                          default="/cam0/image_raw"/>
    <arg name="cam1_topic"                          default="/cam1/image_raw"/>

    <node pkg="viso_pose" type="kitti_raw_node" name="kitti_data_node" output="screen">
        <param name="data_seq_id" value="$(arg sequence_id)" />
        <param name="data_dir" value="/home/hanwen/data/kitti_raw/2011_09_26/2011_09_26_drive_0$(arg sequence_id)_sync" />
        <param name="delay" value="5" />
        <param name="rate" value="1" />
    </node>

    <node pkg="viso_pose" type="viso_node" name="viso_node" output="screen">
        <param name="viso_px_sigma" value="4" />
        <param name="cam0_topic" value="$(arg cam0_topic)" />
        <param name="cam1_topic" value="$(arg cam1_topic)" />
        <rosparam file="$(arg calib_file)" />
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find viso_pose)/rviz/kitti_raw.rviz" />

</launch>
